### PR TITLE
Intergrate astrovault into dex and staking adapter

### PR DESCRIPTION
**Pull request by Buckram123**
### Checklist

- [x] CI is green.
- [x] Changelog updated.

### CODE REVIEW

File: framework/packages/standards/dex/src/msg.rs [hunk 1]
```diff
@@ -142,5 +142,7 @@ pub enum DexQueryMsg {
     GenerateMessages {
         /// Execute message to generate messages for
         message: DexExecuteMsg,
+        /// Manager Addr generate messages for
```
Comment (CyberHoward) [inline framework/packages/standards/dex/src/msg.rs:145]: Why the manager address?

File: integrations/astrovault-adapter/src/dex.rs [hunk 2]
```diff
@@ -0,0 +1,1113 @@
...
+use abstract_sdk::core::objects::PoolType;
+
+use abstract_sdk::feature_objects::AnsHost;
+
+#[derive(Default)]
+pub struct Astrovault {
+    pub pool_type: Option<PoolType>,
+}
+
+impl Identify for Astrovault {
```
Comment (CyberHoward) [inline integrations/astrovault-adapter/src/dex.rs:13]: Can't we update the `Identity` trait to contain consts? Like this: https://doc.rust-lang.org/reference/items/associated-items.html#associated-constants

The function implementations can then be global.

File: framework/packages/standards/dex/src/msg.rs (hunk 1 above)
Comment (Buckram123) [reply framework/packages/standards/dex/src/msg.rs:145]: Oops, yeah it should be proxy

File: integrations/astrovault-adapter/src/dex.rs
Comment (CyberHoward) [inline integrations/astrovault-adapter/src/dex.rs:1]: damn, great job Misha :)

File: integrations/astrovault-adapter/src/dex.rs (hunk 2 above)
Comment (Buckram123) [reply integrations/astrovault-adapter/src/dex.rs:14]: Yeah, we could do that! Created an issue on Linear

File: integrations/astrovault-adapter/src/dex.rs
Comment (Buckram123) [reply integrations/astrovault-adapter/src/dex.rs:1]: Thanks

File: framework/packages/standards/dex/src/msg.rs (hunk 1 above)
Comment (Buckram123) [reply framework/packages/standards/dex/src/msg.rs]: fixed

File: framework/packages/standards/dex/Cargo.toml [hunk 3]
```diff
@@ -17,8 +17,9 @@ crate-type = ["cdylib", "rlib"]
 [features]
 default = ["export"]
 export = []
-testing = ["dep:cw-orch"]
+testing = ["dep:cw-orch", "cw-orch/daemon"]
```
Comment (CyberHoward) [inline framework/packages/standards/dex/Cargo.toml:20]: Why do we need daemon here?

File: modules/contracts/adapters/cw-staking/Cargo.toml [hunk 4]
```diff
@@ -74,6 +75,9 @@ builds = [
   { name = "sei", features = [
     "astroport",
   ] },
+  { name = "archway", features = [
```
Comment (CyberHoward) [inline modules/contracts/adapters/cw-staking/Cargo.toml:78]: I love this so much...

**Review by CyberHoward — APPROVED** [review]

File: framework/packages/standards/dex/Cargo.toml (hunk 3 above)
Comment (Buckram123) [reply framework/packages/standards/dex/Cargo.toml:20]: It's used in `standards/dex/src/tests.rs`

Comment (Buckram123) [reply framework/packages/standards/dex/Cargo.toml:20]: https://github.com/AbstractSDK/abstract/blob/14b695bf19e1e9082f9dd8ff32e82fbd72d82a46/framework/packages/standards/dex/src/tests.rs#L9-L10

**Comment by CyberHoward** [comment]
clippy?

**Comment by Buckram123** [comment]
> clippy?

Thanks, fixed!

**Pull request merged**